src/libstd/net/ip.rs

```diff
@@ -120,7 +120,14 @@ impl Ipv4Addr {
         self.octets()[0] == 169 && self.octets()[1] == 254
     }
 
+    /// Returns true if this address refers to source hosts on "this"
+    /// network. This property defined by RFC 5735.
+    pub fn is_this_network(&self) -> bool {
+        self.octets()[0] == 0
+    }
```

Comment:
Hm we're already a little unsure about including these sorts of funcitons at all, so perhaps this can be left out for now?